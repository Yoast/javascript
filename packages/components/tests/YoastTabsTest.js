import React from "react";
import renderer from "react-test-renderer";

import YoastTabs from "../YoastTabs";

const items = [
	{
		id: "tab1",
		label: "tab1",
		content: <p>This is some content for tab 1. <a href="#1">focusable element 1</a></p>,
	},
	{
		id: "tab2",
		label: "tab2",
		content: <p>This is some content for tab 2. <a href="#2">focusable element 2</a></p>,
	},
	{
		id: "tab3",
		label: "tab3",
		content: <p>This is some content for tab 3. <a href="#3">focusable element 3</a></p>,
	},
];

test( "the YoastTabs matches the snapshot", () => {
	const component = renderer.create(
		<YoastTabs items={ items } />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the YoastTabs with props matches the snapshot", () => {
	const component = renderer.create(
		<YoastTabs
			tabsTextColor="#0f0"
			tabsFontSize="2em"
			tabsTextTransform="uppercase"
			tabsFontWeight="600"
			tabsBaseWidth="300px"
			items={ items }
			onTabsMounted={ () => {
				return "mounted";
			} }
			onTabSelect={ () => {
				return "selected";
			} }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
