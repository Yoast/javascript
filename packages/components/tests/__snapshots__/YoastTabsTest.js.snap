// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the YoastTabs matches the snapshot 1`] = `
<div
  className="YoastTabs__YoastTabsContainer-sa23ab-0 joxfnn"
>
  <div
    className="react-tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        className="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-0"
        role="tab"
        tabIndex="0"
      >
        tab1
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-2"
        role="tab"
        tabIndex={null}
      >
        tab2
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-4"
        role="tab"
        tabIndex={null}
      >
        tab3
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
      tabIndex="0"
    >
      <p>
        This is some content for tab 1. 
        <a
          href="#1"
        >
          focusable element 1
        </a>
      </p>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      className="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
      tabIndex="0"
    />
    <div
      aria-labelledby="react-tabs-4"
      className="react-tabs__tab-panel"
      id="react-tabs-5"
      role="tabpanel"
      tabIndex="0"
    />
  </div>
</div>
`;

exports[`the YoastTabs with props matches the snapshot 1`] = `
<div
  className="YoastTabs__YoastTabsContainer-sa23ab-0 fmuIpQ"
>
  <div
    className="react-tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="true"
        className="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-6"
        role="tab"
        tabIndex="0"
      >
        tab1
      </li>
      <li
        aria-controls="react-tabs-9"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-8"
        role="tab"
        tabIndex={null}
      >
        tab2
      </li>
      <li
        aria-controls="react-tabs-11"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-10"
        role="tab"
        tabIndex={null}
      >
        tab3
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-6"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-7"
      role="tabpanel"
      tabIndex="0"
    >
      <p>
        This is some content for tab 1. 
        <a
          href="#1"
        >
          focusable element 1
        </a>
      </p>
    </div>
    <div
      aria-labelledby="react-tabs-8"
      className="react-tabs__tab-panel"
      id="react-tabs-9"
      role="tabpanel"
      tabIndex="0"
    />
    <div
      aria-labelledby="react-tabs-10"
      className="react-tabs__tab-panel"
      id="react-tabs-11"
      role="tabpanel"
      tabIndex="0"
    />
  </div>
</div>
`;
